### ISSUE TITLE
[FEATURE] Write templates and share them with the community.

### ISSUE THREAD

**Issue by dannywillems** [ocsigen/eliom#368]
Is there a way for the moment to allow developer to write their own Eliom templates, let other developers to download them and use them?
It could be very interesting and a powerful feature for Eliom to grow.

Here an idea how it could be organized:

### To write a template.

To create a mobile template for a chat: 
* create a git repository (on GitHub for example) named *chat-mobile*.
* add files
* push on the repository *https://github.com/user/eliom-template-chat-mobile*

### To use the template *chat-mobile*.
We can think a new argument added to eliom-distillery like
```
eliom-distillery -add-template chat-mobile https://github.com/user/eliom-template-chat-mobile
```
This command will create a directory *chat-mobile* in `$HOME/.opam/$COMPILER/lib/eliom/templates`, download and copy the files from the repository https://github.com/user/eliom-template-chat-mobile.
The list-templates argument will give now
```
basic.ppx
basic.camlp4
chat-mobile
```
and we can now use the template with
```
eliom-distillery -name chat-mobile-test -template chat-mobile
```

**Comment by Drup** [comment]
Isn't that already possible ? ~~eba~~ ocsigen-start installs a new template.

Currently, you have to manually put it in the right directory, though.
As a general thing, we could have eliom-distillery be a standalone project (or investigate https://github.com/dbuenzli/carcass).

**Comment by dannywillems** [comment]
@Drup: Yes it's possible but it's not very easy for a beginner to know it. Eliom-distillery as a standalone project can be very interesting.

I implemented a very basic version of `-add-template` and `-rm-template` using git. I must be improved so I don't make a PR for the moment.
Currently, I find it's a very good solution because I can work on different templates at the same time and start with feature I want (I don't need to start a new project with the entire list of Ocsigen-start features).

I think templates must be in seperated directories as Eliom tools.
I think Eliom-distillery will be the first thing new user will see (because if you want to use ocsigen-start, you need to use Eliom-distillery) so improve it with this kind of features could be very interesting. It will help to grow the community and help to begin new Eliom projects from scratch.

**Comment by dannywillems** [comment]
When this feature will be implemented (maybe #413), documentation about `.eliomignore` and `.reserve` must be added.
